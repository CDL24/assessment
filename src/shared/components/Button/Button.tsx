import React, { useMemo } from "react";
import { TouchableOpacity, Text, TextStyle, ViewStyle } from "react-native";
import createStyles from "./ButtonStyle";
import { useTheme } from "@react-navigation/native";

type ButtonProps = {
  title: String;
  onPress?: () => void;
  titleStyle?: TextStyle;
  buttonStyle?: ViewStyle;
  disabled?: boolean
};
const Button: React.FC<ButtonProps> = ({
  title,
  buttonStyle,
  titleStyle,
  onPress,
  disabled = false
}) => {
  
  const theme = useTheme();
  const styles = useMemo(() => createStyles(theme), [theme]);

  return (
    <TouchableOpacity disabled={disabled} style={[styles.button, buttonStyle]} onPress={onPress}>
      <Text style={[styles.title, titleStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default React.memo(Button);
