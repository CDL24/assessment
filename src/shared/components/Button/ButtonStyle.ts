import { StyleSheet, TextStyle, ViewStyle } from "react-native";
import { ExtendedTheme } from "@react-navigation/native";

interface Style {
  button: ViewStyle;
  title: TextStyle;
}

export default (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create<Style>({
    button: {
      backgroundColor: colors.primary,
      paddingVertical: 16,
      paddingHorizontal: 20,
      borderRadius: 5,
      alignItems: "center",
    },
    title: {
      fontSize: 16,
      color: colors.white,
      fontWeight: "bold",
      justifyContent: "center",
      alignItems: "center",
    },
  });
};
