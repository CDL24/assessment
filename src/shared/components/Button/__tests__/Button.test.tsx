import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import { create } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import Button from '@shared-components/Button/Button';


describe('<Button /> Component', ()=>{
    it('render correctly',()=>{
        const tree = create(<Button title={"Click"} onPress={()=>{}} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('onPress called when click', ()=>{
        const onPress = jest.fn();
        const {getByText} = render(<Button title={"Click"} onPress={onPress} />);
        fireEvent.press(getByText('Click'));
        expect(onPress).toHaveBeenCalled();
    })
});