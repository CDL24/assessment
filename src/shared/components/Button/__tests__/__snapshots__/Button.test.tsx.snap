// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(0, 122, 255)",
      "borderColor": "rgb(0, 122, 255)",
      "borderRadius": 8,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 40,
      "paddingVertical": 23,
    }
  }
>
  <Text
    style={
      [
        {
          "alignItems": "center",
          "color": undefined,
          "fontFamily": "Poppins-Bold",
          "fontSize": 24,
          "fontWeight": "600",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    Click
  </Text>
</View>
`;
